apps/design-tokens-ui/src/pages/theme-detail/components/distribution-curve/distribution-curve.component.ts

```diff
@@ -108,17 +93,10 @@ export class ConstrastDistributionCurveComponent
           this._renderCanvas();
         }
       });
-
-    this._stateChanges$
-      .pipe(debounceTime(700))
-      .pipe(takeUntil(this._destroy$))
-      .subscribe(() => {
-        this.distributionsChange.emit(this._calculateDistributions());
-      });
   }
 
-  ngOnChanges(): void {
-    this._stateChanges$.next();
+  ngOnChanges(changes: SimpleChanges): void {
+    this._stateChanges$.next(changes.generationOptions.currentValue);
```

Comment:
You could check if generationOptions actually changed here, and only next it if it is different.